schema-templates/basic-auth.sql

```diff
@@ -0,0 +1,291 @@
+-------------------------------------------------------------------------------
+-- Adapted from https://github.com/robconery/pg-auth
+
+begin;
+
+create extension if not exists pgcrypto;
+create extension if not exists "uuid-ossp";
+create schema if not exists basic_auth;
+
+-------------------------------------------------------------------------------
+-- Utility functions
+
+create or replace function
+basic_auth.clearance_for_role(u name) returns void as
+$$
+declare
+  ok boolean;
+begin
+  select exists (
+    select rolname
+      from pg_authid
+     where pg_has_role(current_user, oid, 'member')
+       and rolname = u
+  ) into ok;
+  if not ok then
+    raise invalid_password using message =
+      'current user not member of role ' || u;
+  end if;
+end
+$$ LANGUAGE plpgsql;
+
+-------------------------------------------------------------------------------
+-- Login storage and constraints
+
+create table if not exists
+basic_auth.logins (
```

Comment:
I would call this "users", If this is to be a base for other systems i think no one will want to have two (users and logins) tables. And since it will be one table, logins will be strange when you do something like "tasks/&select=id,name,logins(*)"